pkg/sql/catalog/descs/kv_descriptors.go

```diff
@@ -153,39 +154,38 @@ func (kd *kvDescriptors) releaseAllDescriptors() {
 
 func (kd *kvDescriptors) getByID(
 	ctx context.Context, txn *kv.Txn, id descpb.ID, mutable bool,
-) (catalog.Descriptor, error) {
-	// Always pick up a mutable copy so it can be cached.
-	// TODO (lucy): If the descriptor doesn't exist, should we generate our
-	// own error here instead of using the one from catalogkv?
-	desc, err := catalogkv.MustGetMutableDescriptorByID(ctx, txn, kd.codec, id)
-	if err != nil {
-		return nil, err
-	}
-	ud, err := kd.addUncommittedDescriptor(desc)
-	if err != nil {
-		return nil, err
-	}
-	if mutable {
-		return desc, nil
-	}
-	return ud.immutable, nil
+) (desc catalog.Descriptor, err error) {
+	_, desc, err = kd.getDescriptor(ctx, txn, id, mutable, "" /* maybeLookedUpName */)
```

Comment:
I've factored the per-name and per-id descriptor retrieval code in `getDescriptor`.